File: sdk/java/src/main/java/com/aligntech/sdk/FlagConfig.java
package com.aligntech.sdk;

import java.util.Collections;
import java.util.Map;


public class FlagConfig {

    private final String flagKey;
    private final String flagType;
    private final String status;
    private final Map<String, Object> rollout;

    public FlagConfig(String flagKey, String flagType, String status, Map<String, Object> rollout) {
        this.flagKey = flagKey;
        this.flagType = flagType;
        this.status = status;
        this.rollout = rollout != null ? Collections.unmodifiableMap(rollout) : Collections.emptyMap();
    }

    public String getFlagKey() { return flagKey; }
    public String getFlagType() { return flagType; }
    public String getStatus() { return status; }
    public Map<String, Object> getRollout() { return rollout; }

    public boolean isActive() { return "active".equals(status); }
}

File: sdk/dotnet/src/AlignTech.FeatureSDK/Models/FlagConfig.cs
using System.Text.Json.Serialization;

namespace AlignTech.FeatureSDK.Models;

public class FlagConfig
{
    [JsonPropertyName("flagKey")]
    public required string FlagKey { get; set; }

    [JsonPropertyName("flagType")]
    public required string FlagType { get; set; }

    [JsonPropertyName("status")]
    public required string Status { get; set; }

    [JsonPropertyName("rollout")]
    public Dictionary<string, object>? Rollout { get; set; }
}
